TLM/TLM/Manager/Impl/OptionsManager.cs

```diff
@@ -309,7 +309,7 @@ private static void ToSlider(byte[] data, uint idx, ILegacySerializableOption op
         /// Triggers a rebuild of the main menu (toolbar), adding/removing buttons
         /// where applicable, and refreshing all translated text. Very slow.
         /// </summary>
-        internal static void RebuildMenu() {
+        internal static void RebuildMenu(bool reloadTextures = false) {
```

Comment:
I think this would be better named `languageChanged` as that's more specific to what happened.

Note that the `TMPELifecycle.Instance.TranslationDatabase` methods called later in the method are not texture related. I have no idea why they need treating separately from all the other locale strings.

Reply:
Considering there's only 2 calls that set the `bool` to `true` (and they are both in same method - that handles language change) is it worth splitting out that block to separate function as it doesn't really have anything to do with rebuilding main menu.

Reply:
True, but now I read the code and I don't see how it works when main game language changes. There is no trigger... 🤷‍♂️

Reply:
`KeybindSettingsBase:OnLocaleChanged` is this something of interest?

Reply:
@kvakvs you mean this code that does nothing? 🧐 
![image](https://user-images.githubusercontent.com/19638970/161641912-42bc85af-7de0-4b14-b242-6224770b64a7.png)

Reply:
It subscribes to locale change in localemanager, and that's assembly-csharp, the C:S DLL

Reply:
Ah, yeah, there's two places language can change - the mod option (global config based) and if that's `null` then we look at the game language IIRC so need to listen for vanilla game language change (and have some conversion between its language codes, which can be modded, to our language codes).

@krzychu124 I did a load of digging in https://github.com/CitiesSkylinesMods/TMPE/issues/1379 and traced what I found in comments there which might be of some use to getting idea of how the locale stuff all fits together.

Reply:
> It subscribes to locale change in localemanager, and that's assembly-csharp, the C:S DLL

`LocaleManager` is in `ColossalManaged.dll` but that doesn't even matter here.
@kvakvs that code is in `KeybindSettingsBase` and method `OnLocaleChanged` does literally nothing. Body in our method is empty and you probably meant `OnLocaleChanged` but from the `UIComponent` (`KeybindSettingsBase` extends `UICustomControl` **which is not** `UIComponent` - `UICustomControl` only hold reference to `UIComponent`). 

It looks like a bug or some leftovers after refactoring.

Reply:
something seems to be wrong....

now the UI disapears if the pathfind is turned off :D

![image](https://user-images.githubusercontent.com/37211016/161827264-2d2d2503-56d0-4802-af75-f88ea1151d89.png)

Reply:
So final Test... With the correct DLL 😁

It works!

Reply:
@krzychu124 does the stuff about empty `OnLocaleChanged` need tracking in separate issue?

otherwise this conversation block can be marked resolved

Reply:
I would create new issue and we will see what to do with that thing (what was the intention and if is there any bug). This conversation can be marked as resolved.

Reply:
Created issue: https://github.com/CitiesSkylinesMods/TMPE/issues/1520